src/perform_step/ssprk_perform_step.jl

```diff
@@ -51,6 +51,66 @@ end
   f( k,  u, p, t+dt)
 end
 
+function initialize!(integrator,cache::KYKSSPRK42Cache)
+  @unpack k,fsalfirst = cache
+  integrator.fsalfirst = fsalfirst
+  integrator.fsallast = k
+  integrator.kshortsize = 1
+  resize!(integrator.k, integrator.kshortsize)
+  integrator.k[1] = integrator.fsalfirst
+  integrator.f(integrator.fsalfirst,integrator.uprev,integrator.p,integrator.t) # FSAL for interpolation
+end
+
+@muladd function perform_step!(integrator,cache::KYKSSPRK42Cache,repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack k,tmp,u2,fsalfirst = cache
+  @unpack α10, α20, α21, α30, α32, α40, α43, β10, β21, β30, β32, β40, β43 = cache.tab
+
+  # u1 -> stored as u
+  @. u = α10*uprev + dt*β10*integrator.fsalfirst
+  f(u, p, t)
+  # u2
+  @. u₂ = α20*uprev + α21*u + dt*β21*k
+  f(u₂, p, t)
+  # u3
+  @. tmp = α30*uprev + α32*u2 + dt*β30*integrator.fsalfirst + dt*β32*k
```

Comment:
you might want to `@..` this line and the next one.

Reply:
It's a special internal macro we made which is more optimized than `@.` on long expressions.

Reply:
Ok. I'll change it to @..

The paper is Kubatko, Yeager, Ketcheson (2014)